### ISSUE TITLE
Primární identifikátor v RDF

### ISSUE THREAD

**Issue by vbar** [ARUP-CAS/arup-da-amcr#104]
Mapování AMCR záznamů do CRM generuje RDF graf s různými entitami, a není zřejmé která z nich odpovídá OAI-PMH záznamu - např. serializace externího zdroje může začínat

<pre>
&lt;rdf:RDF ...
  &lt;crm:E33_Linguistic_Object rdf:about="https://api.aiscr.cz/id/X-BIB-0710038-part9">
</pre>

ačkoli záznam fakticky nepopisuje https://api.aiscr.cz/id/X-BIB-0710038-part9 , ale crm:E31_Document https://api.aiscr.cz/id/X-BIB-0710038 (viz též #97). Problém je, že X3ML transformace mapuje crm:E31_Document (mj.) do pozice objektu RDF tripletu:

<pre>
        &lt;mapping>
            &lt;domain>
                &lt;source_node>/amcr/ext_zdroj/ext_odkaz&lt;/source_node>
                &lt;target_node>
                    &lt;entity>
                        &lt;type>crm:E33_Linguistic_Object&lt;/type>
...
                &lt;range>
                    &lt;source_node>externi_zdroj&lt;/source_node>
                    &lt;target_node>
                        &lt;entity>
                            &lt;type>crm:E31_Document&lt;/type>
</pre>

Tuto definici by bylo vhodné invertovat (viz přiložený soubor).

**Comment by motyc** [comment]
Tak jsem tomu teď věnoval poměrně dost času a zatím se mi nedaří najít v rámci mappingu takové řešení, aby produkovalo validní výsledky. Budu nad tím ale ještě dumat. Myslím, že jsem nevyčerpal všechny možnosti.

**Comment by motyc** [comment]
Tak jsem nad tím snad vyhrál. Sice to znamenalo nějaké ústupky, ale nic fatálního... Zjistil jsem, že mapping nezvládá, aby se tam jeden atribut použil dvakrát a tím pádem jsem musel něco vyřadit, aby to fungovalo, ale fakticky jsme o nic nepřišli (ztratila se vlastnost P48 u podřazených tříd, ale to nevadí). Snad je to nyní mapované dobře.

[AMCR_to_CIDOC_X3ML_190604.zip](https://github.com/ARUP-CAS/arup-da-amcr/files/3252848/AMCR_to_CIDOC_X3ML_190604.zip)

Ještě si říkám, zda nebude problém, že nemáme v mapování datum u dokumentační jednotky. Nebude to také problém kvůli `<datestamp>`?

**Comment by vbar** [comment]
Ano, to vypadá dobře. Co se týče datumu u dokumentační jednotky, chybějící datestamp by byl problém (OAI-PMH v zásadě očekává že každý záznam má datestamp), ale jelikož už jsme datum přidali do interního XML (element &lt;datum&gt;), nedalo by se prostě namapovat i do RDF?

**Comment by motyc** [comment]
Ano, dalo, proto se na to ptám :) Dodělám to a pošlu.

**Comment by motyc** [comment]
Zde je aktualizace. Udělal jsem tam i nějaké změny jako přípravu kvůli #112, ale mělo by to vše fungovat při současném vstupu.
